extensions/rvadapter/src/main/java/com/google/android/agera/rvadapter/RepositoryAdapter.java

```diff
@@ -128,7 +133,84 @@ public RepositoryAdapter build() {
       return new RepositoryAdapter(this);
     }
 
+    /**
+     * Builds the {@link RepositoryAdapter} that presents the provided repositories in order and
+     * observes the repositories as well as any additional observables while the provided
+     * {@link Activity} is active (between onResumed/onPaused).
+     */
+    @TargetApi(14)
+    @NonNull
+    public Adapter<ViewHolder> whileActive(@NonNull final Activity activity) {
+      final RepositoryAdapter repositoryAdapter = new RepositoryAdapter(this);
+      activity.getApplication().registerActivityLifecycleCallbacks(
+          new AdapterActivityLifecycleCallbacks(){
+            @Override
+            public void onActivityResumed(final Activity activity) {
+              repositoryAdapter.startObserving();
```

Comment:
you need to wrap the (outer parameter) activity into a WeakReference and check if this (inner parameter) activity is the same instance before you start/stop observing.